From 1b118b9fbb7b2fbcebb164d00f26d120e4c98252 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 26 Dec 2011 22:50:22 -0600
Subject: [PATCH 135/154] pvrvideosink: fix problem with non-ducati buffers

In some cases dvdspu seems to give us a non-pad-alloc'd buffer.  Don't
crash randomly in these cases.
---
 sys/pvr2d/gstpvrvideosink.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index abb911f..c2c4d6a 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -1154,7 +1154,7 @@ gst_pvrvideosink_show_frame (GstBaseSink * vsink, GstBuffer * buf)
 
   GST_DEBUG_OBJECT (pvrvideosink, "render buffer: %p", buf);
 
-  if (!GST_IS_BUFFER (buf)) {
+  if (!GST_IS_DUCATIBUFFER (buf)) {
     GstFlowReturn ret;
 
     /* special case check for sub-buffers:  In certain cases, places like
-- 
1.7.5.4

